saleor/checkout/complete_checkout.py

```diff
@@ -1053,21 +1053,39 @@ def _handle_allocations_of_order_lines(
     )
 
 
-def _handle_checkout_discount(order: "Order", checkout: "Checkout"):
+def _create_order_discount(order: "Order", checkout_info: "CheckoutInfo"):
+    checkout = checkout_info.checkout
     if checkout.discount:
-        # store voucher as a fixed value as it this the simplest solution for now.
-        # This will be solved when we refactor the voucher logic to use .discounts
-        # relations
-
-        order.discounts.create(
-            type=DiscountType.VOUCHER,
-            value_type=DiscountValueType.FIXED,
-            value=checkout.discount.amount,
-            name=checkout.discount_name,
-            translated_name=checkout.translated_discount_name,
-            currency=checkout.currency,
-            amount_value=checkout.discount_amount,
-        )
+        checkout_discount = checkout.discounts.first()
+        if checkout_discount and checkout_discount.type == DiscountType.ORDER_PROMOTION:
+            order.discounts.create(
+                type=checkout_discount.type,
+                value_type=checkout_discount.value_type,
+                value=checkout_discount.value,
+                name=checkout_discount.name,
+                translated_name=checkout_discount.translated_name,
+                currency=checkout_discount.currency,
+                amount_value=checkout_discount.amount_value,
+                promotion_rule=checkout_discount.promotion_rule,
+            )
```

Comment:
As we already have the object that is almost the same. We can do something like that:
```
discount_data = model_to_dict(checkout_discount)
order.discount.create(**discount_data)
```